### PR TITLE
[Improvement][Security]Add CSRF protection(issue-12931)

### PR DESCRIPTION
**Pull request by hdygxsj**
## Purpose of the pull request

close issue #12931

Currently, the dolphinscheduler can be attacked by a CSRF in the following ways.

1. First, someone exits the Dolphinscheduler site but does not clear the cookie
2. The dolphinscheduler api can then be accessed via cookies by other sites or others to submit forms or obtain information without authorization. For example:
```tsx
import { defineComponent, ref } from "vue";

export default defineComponent({
  setup() {
  },
  render() {
    return (<div><form action="http://127.0.0.1:5173/dolphinscheduler/projects" method="post">
      <input type="hidden"
        name="projectName"
        value="dsadasdddddddd" />
      <input type="hidden"
        name="userName"
        value="admin" />
      <input type="submit"
        value="Win Money!" />
    </form></div>)
  }
})
```

Once the dolphinscheduler user clicks the button in the diagram, a csrf attack completes

after this pr, when an attacker performs a CSRF attack, he will get a 403 exception.

## Brief change log

I thought it would take a lot of work to introduce Spring security dependencies, so I added a CsrfTokenInterceptor to implement csrf defense.
The CsrfTokenInterceptor does not intercept requests with the token in the http request header to ensure that the api is accessible through the token.
CsrfTokenInterceptor does not intercept http get requests.
The CsrfTokenInterceptor will get the X-CSRF-TOKEN from the http request header or the _csrf from the request paramter and verify their accuracy.
And changed the ui module to add the request header X-CSRF-TOKEN when making an http request
With the above approach, the dolphinscheduler is protected from CSRF attacks as long as the attacker cannot obtain the correct CSRF token.
In this pr, the csrfToken is obtained by inverting the sessionId.   If we need to, I can enhance the protection of the csrfToken, such as using the encryption algorithm or using other means to generate our csrfToken

### CODE REVIEW

**Review by SbloodyS — COMMENTED** [review]
Please check the failed CI.

**Comment by hdygxsj** [comment]
> Please check the failed CI.

done

File: dolphinscheduler-ui/src/service/service.ts [hunk 1]
```diff
@@ -74,9 +74,12 @@ const err = (err: AxiosError): Promise<AxiosError> => {
 service.interceptors.request.use((config: AxiosRequestConfig<any>) => {
   config.headers && (config.headers.sessionId = userStore.getSessionId)
   const language = cookies.get('language')
+  const sessionId = cookies.get('sessionId')
   config.headers = config.headers || {}
   if (language) config.headers.language = language
-
+  if (sessionId) {
+    config.headers['X-CSRF-TOKEN'] = sessionId.split('').reverse().join('')
```
Comment (EricGao888) [inline dolphinscheduler-ui/src/service/service.ts:81]: Currently, you invert the sessionId to get a `CSRF` token and put it in headers in front-end. As you wrote in PR change log, you plan to encrypt it in the future, may I ask where and how you plan to encrypt it?

Comment (hdygxsj) [reply dolphinscheduler-ui/src/service/service.ts:81]: My current thoughts are as follows,
1. Use the asymmetric encryption algorithm to encrypt. When the front-end end obtains the sessionId, encrypt it using the public key and store the ciphertext persistently using pinia. When the back-end interceptor receives the X-CSRF-TOKEN, decrypt it with the private key and compare it with the sessionId. This comes at the cost of reducing the throughput of the request.
2. Or the back end can provide a get api for obtaining a token. The back end generates the token and persists it in the database. However, I feel that the io cost of querying the database should be higher than that of decrypting.

If the above methods are OK, I can also implement it in this pr

Comment (EricGao888) [reply dolphinscheduler-ui/src/service/service.ts:81]: WDYT? @kezhenxu94 @caishunfeng @SbloodyS

**Comment by EricGao888** [comment]
Hi @hdygxsj , DS community holds an online bi-weekly meeting. You are very welcome to bring up this topic for discussion during the meeting if you'd like to. Thanks : )

Here is the link:
https://docs.qq.com/doc/DQ3BGVHZ1bXp5R2FZ

**Comment by hdygxsj** [comment]
> Hi @hdygxsj , DS community holds an online bi-weekly meeting. You are very welcome to bring up this topic for discussion during the meeting if you'd like to. Thanks : )
> 
> Here is the link: https://docs.qq.com/doc/DQ3BGVHZ1bXp5R2FZ

I will be attending the meeting. In addition, I added the implementation using spring security, see pr #13235

**Comment by kezhenxu94** [comment]
Close in favor of https://github.com/apache/dolphinscheduler/pull/13235, 13235 would be a more neat implementation

**Pull request closed without merging**